### PR TITLE
Using skt.misc instead of constant number for retcode

### LINKED ISSUE #287: Use constants for success/fail/error instead of 0/1/2
As @danghai pointed out in one of the pulls, it would be more clear to understand than plain numbers.

Implement some simple constants like `SKT_SUCCESS = 0`, `SKT_FAIL = 1`, `SKT_ERROR = 2`, and then use these across the code instead of the direct numeric values where required.

### PR DESCRIPTION
**Pull request by danghai**
I make PR to use `skt.misc` instead of constant number for retcode.
Fix #287

### CODE REVIEW

**Comment by danghai** [comment]
@veruu So you want me to remove the class Result. Just declare the constants. It probably uses `skt.misc.SKT_` ?

**Comment by veruu** [comment]
Yes, just declaring the constants would be much easier and understandable. And then do `from sktm.misc import <LIST OF CONSTANTS>` and use the constant name directly, instead of using the `skt.misc` prefix on each line (since the prefix itself is not very useful)

**Comment by danghai** [comment]
@veruu I have fixed it.

File: skt/runner.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 import subprocess
 import time
 import xml.etree.ElementTree as etree
+from skt.misc import SKT_SUCCESS, SKT_FAIL, SKT_ERROR
```
Comment (veruu) [inline skt/runner.py]: Can you add a newline before this line, to divide the import groups?

File: skt/runner.py [hunk 2]
```diff
@@ -135,7 +136,7 @@ def dumpjunitresults(self, jobid, junit):
         with open(fname, 'w') as fileh:
             bkr = subprocess.Popen(args, stdout=fileh)
             bkr.communicate()
-        if bkr.returncode != 0:
+        if bkr.returncode != SKT_SUCCESS:
```
Comment (veruu) [inline skt/runner.py]: This one is a subprocess return code which shouldn't be compared with skt results.

File: skt/runner.py [hunk 3]
```diff
@@ -292,30 +294,31 @@ def __getresults(self, jobid=None):
...
-            infrastructure failures.
+            SKT_SUCCESS if all jobs passed,
+            SKT_FAIL in case of failures, and
+            SKT_ERROR in case of infrastructure failures.
         """
-        ret = 0
+        ret = SKT_SUCCESS
         fhosts = set()
-        tfailures = 0
+        tfailures = SKT_SUCCESS
```
Comment (veruu) [inline skt/runner.py]: This variable represents number of test failures and not a return code

File: skt/runner.py [hunk 4]
```diff
@@ -397,7 +400,7 @@ def __watchloop(self):
                                       ' rest of running jobs and aborting!')
                         self.failures[cid] = [[root.attrib.get('system')],
                                               set([(result, status)]),
-                                              1]
+                                              SKT_FAIL]
```
Comment (veruu) [inline skt/runner.py]: Again here, the last element of the list represents number of failed test runs and not the return code. Yes, we need to clean up the runner :)

File: skt/runner.py [hunk 5]
```diff
@@ -432,7 +435,7 @@ def __watchloop(self):
                                 origin = cid
 
                             if origin not in self.failures:
-                                self.failures[origin] = [[], set(), 1]
+                                self.failures[origin] = [[], set(), SKT_FAIL]
```
Comment (veruu) [inline skt/runner.py]: Same as above

File: skt/runner.py [hunk 6]
```diff
@@ -528,15 +531,15 @@ def run(self, url, release, wait=False, host=None, uid="",
...
+            SKT_FAIL if testing failed
+            SKT_ERROR in case of infrastructure error (exceptions are logged)
         """
-        ret = 0
+        ret = SKT_SUCCESS
         self.failures = {}
         self.recipes = set()
         self.watchlist = set()
-        self.aborted_count = 0
+        self.aborted_count = SKT_SUCCESS
```
Comment (veruu) [inline skt/runner.py]: As the variable name suggests, the variable contains a count of aborted jobs, not a return code

File: skt/kerneltree.py [hunk 7]
```diff
@@ -334,7 +335,7 @@ def merge_patchwork_patch(self, uri):
         status, output = self.__git_cmd_pipe(patch_content,
                                              "am", "-",
                                              cwd=self.wdir)
-        if status != 0:
+        if status != SKT_SUCCESS:
```
Comment (veruu) [inline skt/kerneltree.py]: Status here is the return code from subprocess call, so this one shouldn't be compared with the result codes

File: skt/executable.py [hunk 8]
```diff
@@ -31,15 +31,17 @@
 import junit_xml
 
 import skt
+import skt.misc
```
[the remark this answers is not in the document]
Comment (danghai) [reply skt/executable.py]: I forget removing it.

**Comment by danghai** [comment]
@veruu I fixed it!

File: skt/executable.py [hunk 9]
```diff
@@ -34,12 +34,14 @@
 import skt.publisher
 import skt.reporter
 import skt.runner
+
```
[the remark this answers is not in the document]
Comment (danghai) [reply skt/executable.py]: Oh sorry, I think it is a import group. That is my mistake. I fix now

File: skt/kerneltree.py [hunk 10]
```diff
@@ -20,6 +20,8 @@
 
 import skt
 
+from skt.misc import SKT_SUCCESS, SKT_FAIL
```
[the remark this answers is not in the document]
Comment (danghai) [reply skt/kerneltree.py]: Do you mean no new line between `import skt` and `from skt.misc import SKT_SUCCESS, SKT_FAIL`?

[the remark this answers is not in the document]
Comment (veruu) [reply skt/kerneltree.py]: Yes. In general, there are 3 import groups: standard modules, third party modules, and local modules. skt is a local module, so both skt imports should be together in the third group.

[the remark this answers is not in the document]
Comment (danghai) [reply skt/kerneltree.py]: Oh thanks. I never know about this standard before. I fixed it

[the remark this answers is not in the document]
Comment (veruu) [reply skt/kerneltree.py]: You removed the wrong newline :) the newline in runner should be there, the one that should have been removed is in this kerneltree file :)

**Comment by danghai** [comment]
@veruu I fixed it!

**Comment by danghai** [comment]
@veruu Thanks. I fixed it.

**Pull request merged**